marlowe-actus/src/Language/Marlowe/ACTUS/Model/INIT/StateInitializationModel.hs

```diff
@@ -56,3 +59,60 @@ _INIT_PAM t0 tminus tfp_minus tfp_plus _MD _IED _IPNR _CNTRL _NT _IPAC _DCC _FER
         sd                                      = t0
     in ContractStatePoly { prnxt = 0.0, ipcb = 0.0, tmd = tmd, nt = nt, ipnr = ipnr, ipac = ipac, fac = fac, feac = feac, nsc = nsc, isc = isc, prf = prf, sd = sd }
 
+_INIT_LAM scfg t0 tminus tpr_minus tfp_minus tfp_plus _MD _IED _IPNR _CNTRL _NT _IPAC _DCC _FER _FEAC _FEB _SCEF _SCIXSD _PRF _PRCL _PRANX _PRNXT _IPCB _IPCBA =
+    let
+        tmd = (paymentDay . head . fromJust $ _SCHED_MD_LAM scfg _IED t0 _PRANX _PRNXT _NT _PRCL _MD)
```

Comment:
Since the spec references the schedule value here, we call the schedule function and get the payment day of the returned ShiftedDay.

Reply:
> 2. initial state of tmd is expressed differently in this version of spec: https://www.actusfrf.org/techspecs

True, I realized I had defined tmd initial state in terms of the schedule, where it should have been the other way around. Once fixed that things got a lot cleaner and the weird stuff with the shifting here went away.